app/javascript/packages/document-capture/hooks/use-log-camera-info.ts

```diff
@@ -0,0 +1,69 @@
+import { useState, useEffect, useContext } from 'react';
+import AnalyticsContext from '../context/analytics';
+
+function getConstraints(deviceId) {
+  return {
+    video: {
+      width: {
+        ideal: 999999,
+      },
+      height: {
+        ideal: 999999,
+      },
+      deviceId: {
+        exact: deviceId,
+      },
+    },
+  };
+}
+
+function getCameraInfo(videoTrack) {
+  const cameraInfo = {
+    label: videoTrack.label,
+    frameRate: videoTrack.getSettings().frameRate,
+    height: videoTrack.getSettings().height,
+    width: videoTrack.getSettings().width,
+  };
+  return cameraInfo;
+}
+
+async function updateConstraintsAndLogInfo(videoDevice, trackEvent) {
+  // See https://developer.mozilla.org/en-US/docs/Web/API/MediaTrackConstraints/facingMode
+  const updatedConstraints = getConstraints(videoDevice.deviceId);
+  try {
+    const stream = await navigator.mediaDevices.getUserMedia(updatedConstraints);
+    const videoTracks = stream.getVideoTracks();
+    const cameras = videoTracks.map((videoTrack) => getCameraInfo(videoTrack));
+    const logInfo = {
+      camera_info: cameras,
+    };
+    trackEvent('IdV: camera info logged', logInfo);
+  } catch (error) {
+    trackEvent('IdV: camera info error', error);
+  }
+}
+
+async function logCameraInfo(trackEvent) {
+  const devices = await navigator.mediaDevices.enumerateDevices();
+  const videoDevices = devices.filter((device) => device.kind === 'videoinput');
+  videoDevices.forEach((videoDevice) => {
+    updateConstraintsAndLogInfo(videoDevice, trackEvent);
+  });
+}
+
+function useLogCameraInfo({ isBackOfId, hasStartedCropping }) {
+  const [didLogCameraInfo, setDidLogCameraInfo] = useState(false);
+  const { trackEvent } = useContext(AnalyticsContext);
+
+  useEffect(() => {
+    if (!isBackOfId) {
+      return;
+    }
+    if (hasStartedCropping && !didLogCameraInfo) {
+      logCameraInfo(trackEvent);
+      setDidLogCameraInfo(true);
```

Comment:
I'd wonder if we should track `didLogCameraInfo` as a mutable `useRef` value instead of state, to (a) avoid quirks of [state batching](https://react.dev/learn/queueing-a-series-of-state-updates) that might result in this happening asynchronously, and (b) to avoid triggering an unnecessary re-render on state update.

Reply:
Oh interesting, I'm not as familiar with `useRef` but this is a great use case for that. I [replaced the state ](https://github.com/18F/identity-idp/pull/10227/commits/3582deaa28d03bb8671b3d3c51fe9634229b318c)with a ref.